src/main/java/bsh/BSHFormalParameter.java

```diff
@@ -62,9 +62,24 @@ public String getTypeDescriptor(
     public Object eval( CallStack callstack, Interpreter interpreter)
         throws EvalError
     {
-        if ( jjtGetNumChildren() > 0 )
+        if ( jjtGetNumChildren() > 0 ) {
             type = ((BSHType)jjtGetChild(0)).getType( callstack, interpreter );
-        else
+            /*
+             * Check if dimensions have been recorded for this parameter.
+             * If so,
+             * - If no array dimensions on the type then add them now.
+             * - If there are already array dimensions on the type then
+             * throw an error.
+             */
+            if (dimensions > 0) {
+                if (!type.isArray())
+                    type = Array.newInstance(type, new int[dimensions]).getClass();
+                else
+                    throw new EvalError(
+                       "Array dimensions not allowed on both type and name: "
+                       + name, this, null );
```

Comment:
This seems like an easy enough test to add.

Reply:
Codecov flagged as untested.

Reply:
Added test